api/main.go

```diff
@@ -582,7 +585,40 @@ func (serv *MetadataServer) ListProviders(in *pb.Empty, stream pb.Api_ListProvid
 
 func (serv *MetadataServer) CreateProvider(ctx context.Context, provider *pb.Provider) (*pb.Empty, error) {
 	serv.Logger.Infow("Creating Provider", "name", provider.Name)
-	return serv.meta.CreateProvider(ctx, provider)
+	_, err := serv.meta.CreateProvider(ctx, provider)
+	if err != nil {
+		serv.Logger.Errorw("Failed to create provider", "error", err)
+		return nil, err
+	}
+	var status *pb.ResourceStatus
+	isHealthy, err := serv.health.Check(provider.Name)
+	if err != nil || !isHealthy {
+		serv.Logger.Errorw("Provider health check failed", "error", err)
+		status = &pb.ResourceStatus{
+			Status:       pb.ResourceStatus_FAILED,
+			ErrorMessage: err.Error(),
+		}
+	} else {
+		serv.Logger.Infow("Provider health check passed", "name", provider.Name)
+		status = &pb.ResourceStatus{
+			Status: pb.ResourceStatus_READY,
+		}
+	}
+	statusReq := &pb.SetStatusRequest{
+		ResourceId: &pb.ResourceID{
+			Resource: &pb.NameVariant{
+				Name: provider.Name,
+			},
+			ResourceType: pb.ResourceType_PROVIDER,
+		},
+		Status: status,
+	}
+	_, statusErr := serv.meta.SetResourceStatus(ctx, statusReq)
+	if statusErr != nil {
+		serv.Logger.Errorw("Failed to set provider status", "error", statusErr, "health check error", err)
+		return nil, statusErr
+	}
+	return &pb.Empty{}, err
```

Comment:
This logic will now set providers to either a Failed or Ready status based on the results of the health check.

Ready Example (dashboard view):
<img width="858" alt="Screen Shot 2023-10-18 at 9 23 03 PM" src="https://github.com/featureform/featureform/assets/10430775/864107f1-bce8-4690-9800-d0315019a7f1">

Failed Example (dashboard view):
<img width="888" alt="Screen Shot 2023-10-18 at 9 22 00 PM" src="https://github.com/featureform/featureform/assets/10430775/e8b36e7a-1391-4be3-8ec1-a2eb08fd88f9">

Thoughts, @sdreyer?